src/signalcorr.jl

```diff
@@ -61,34 +59,31 @@ where each column in the result will correspond to a column in `x`.
 
 The output is not normalized. See [`autocor!`](@ref) for a method with normalization.
 """
-function autocov!(r::RealVector, x::RealVector, lags::IntegerVector; demean::Bool=true)
+function autocov!(
+    r::AbstractVector,
+    x::AbstractVector,
+    lags::AbstractVector{<:Integer},
+    z=Vector{typeof(zero(eltype(x)) / 1)}(undef, size(x, 1));
+    demean::Bool=true
+    )
     lx = length(x)
     m = length(lags)
     length(r) == m || throw(DimensionMismatch())
     check_lags(lx, lags)
-
-    T = typeof(zero(eltype(x)) / 1)
-    z::Vector{T} = demean ? x .- mean(x) : x
+    demean ? z .= x .- mean(x) : copyto!(z, x)
     for k = 1 : m  # foreach lag value
         r[k] = _autodot(z, lx, lags[k]) / lx
     end
     return r
 end
 
-function autocov!(r::RealMatrix, x::RealMatrix, lags::IntegerVector; demean::Bool=true)
-    lx = size(x, 1)
-    ns = size(x, 2)
-    m = length(lags)
-    size(r) == (m, ns) || throw(DimensionMismatch())
-    check_lags(lx, lags)
-
+function autocov!(
+    r::AbstractMatrix, x::AbstractMatrix, lags::AbstractVector{<:Integer}; demean::Bool=true
+    )
     T = typeof(zero(eltype(x)) / 1)
-    z = Vector{T}(undef, lx)
-    for j = 1 : ns
-        demean_col!(z, x, j, demean)
-        for k = 1 : m
-            r[k,j] = _autodot(z, lx, lags[k]) / lx
-        end
+    z = Vector{T}(undef, size(x, 1))
+    for n in 1:size(x, 2)
+        autocov!(view(r, :, n), view(x, :, n), lags, z; demean)
     end
```

Comment:
Why did you replace the old version? It seems fine and also did not cause unnecessary allocations - vector `z` was allocated up-front and reused for every column.